components/web_discovery/browser/background_credential_helper_impl.cc

```diff
@@ -0,0 +1,124 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/browser/background_credential_helper_impl.h"
+
+#include <utility>
+
+#include "base/base64.h"
+#include "base/containers/span_rust.h"
+#include "base/logging.h"
+#include "base/threading/thread_restrictions.h"
+#include "crypto/sha2.h"
+
+namespace web_discovery {
+
+BackgroundCredentialHelperImpl::BackgroundCredentialHelperImpl()
+    : anonymous_credentials_manager_(new_anonymous_credentials_manager()) {}
+
+BackgroundCredentialHelperImpl::~BackgroundCredentialHelperImpl() = default;
+
+void BackgroundCredentialHelperImpl::UseFixedSeedForTesting() {
+  anonymous_credentials_manager_ = new_anonymous_credentials_with_fixed_seed();
+}
+
+std::unique_ptr<RSAKeyInfo> BackgroundCredentialHelperImpl::GenerateRSAKey() {
+  auto key_pair = GenerateRSAKeyPair();
+  if (!key_pair) {
+    return nullptr;
+  }
+  rsa_private_key_ = std::move(key_pair->private_key);
```

Comment:
I'm confused here, why are we generating an `RSAKeyInfo`, moving the private key to a class member and then returning the keypair that no longer contains the private key?